undertow/src/main/java/org/wildfly/extension/undertow/UndertowTransformers.java

```diff
@@ -118,13 +120,20 @@ private static void registerTransformersWildFly18(ResourceTransformationDescript
     }
 
     private static void registerTransformersWildFly16(ResourceTransformationDescriptionBuilder subsystemBuilder) {
+        subsystemBuilder.getAttributeBuilder().addRejectCheck(RejectAttributeChecker.DEFINED, OBFUSCATE_SESSION_ROUTE)
+                .setDiscard(DiscardAttributeChecker.DEFAULT_VALUE, OBFUSCATE_SESSION_ROUTE)
+                .end();
```

Comment:
@fl4via I believe you need a new 'registerTransformersWildFly22 method (and a MODEL_VERSION_WILDFLY_22) and this code goes there.

It then shouldn't be necessary to repeat this logic in the other methods. You handle it in current->MODEL_VERSION_WILDFLY_22 and the chain transformation means its handled for all the earlier ones.

Reply:
A code style nit is that it's nicer to do the setDiscard call before the addRejectCheck call. The order of these doesn't matter technically as the transformation engine does discards before rejects. But maintainers are less likely to know that and putting the code in the order they actually happen helps make it clearer what will happen.

I wouldn't mention this if it weren't for that fact that my comment above means this will be getting touched anyway.

Reply:
Sure, @bstansberry  I'll update the PR